### PR TITLE
[MKM] Implement Yarus, Roar of the Old Gods

### PR DESCRIPTION
**Pull request by DominionSpy**
#11516

### CODE REVIEW

File: Mage.Sets/src/mage/cards/y/YarusRoarOfTheOldGods.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        this.toughness = new MageInt(4);
+
+        // Other creatures you control have haste.
+        this.addAbility(new SimpleStaticAbility(new GainAbilityControlledEffect(
+                HasteAbility.getInstance(), Duration.WhileOnBattlefield,
+                StaticFilters.FILTER_PERMANENT_CREATURES, true)));
+        // Whenever one or more face-down creatures you control deal combat damage to a player, draw a card.
+        this.addAbility(new DealCombatDamageControlledTriggeredAbility(
+                Zone.BATTLEFIELD, new DrawCardSourceControllerEffect(1),
+                filter, SetTargetPointer.PLAYER, false)
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/y/YarusRoarOfTheOldGods.java:63]: should be no need to set target pointer here? simpler constructor should work fine?

File: Mage.Sets/src/mage/cards/y/YarusRoarOfTheOldGods.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        Player controller = game.getPlayer(source.getControllerId());
+        Card card = game.getCard(getTargetPointer().getFirst(game, source));
+        if (controller == null || card == null || !card.isPermanent(game)) {
+            return false;
+        }
+
+        Ability newSource = source.copy();
+        newSource.setWorksFaceDown(true);
+        MageObjectReference mor = new MageObjectReference(card.getId(),
+                card.getZoneChangeCounter(game), game);
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/y/YarusRoarOfTheOldGods.java:108]: this feels like the zcc would be off by one but if works in testing then all fine

**Review by xenohedron — APPROVED** [review]

File: Mage.Sets/src/mage/cards/y/YarusRoarOfTheOldGods.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        game.addEffect(new BecomesFaceDownCreatureEffect(
+                null, mor, Duration.Custom,
+                BecomesFaceDownCreatureEffect.FaceDownType.MANUAL), newSource);
+        controller.moveCards(card, Zone.BATTLEFIELD, source, game, false, true, true, null);
+
+        // Need to process the move so that any ETB triggers on the card don't fire
+        game.getState().processAction(game);
+
+        Permanent permanent = game.getPermanent(card.getId());
+        return permanent != null && permanent.turnFaceUp(source, game, source.getControllerId());
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/y/YarusRoarOfTheOldGods.java:116]: Must return true all the time here. True/false logic in most apply’s use cases: if something changed in game state then return true.

File: Mage.Sets/src/mage/cards/y/YarusRoarOfTheOldGods.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        newSource.setWorksFaceDown(true);
+        MageObjectReference mor = new MageObjectReference(card.getId(),
+                card.getZoneChangeCounter(game) + 1, game);
+        game.addEffect(new BecomesFaceDownCreatureEffect(
+                null, mor, Duration.Custom,
+                BecomesFaceDownCreatureEffect.FaceDownType.MANUAL), newSource);
+        controller.moveCards(card, Zone.BATTLEFIELD, source, game, false, true, true, null);
+
+        // Need to process the move so that any ETB triggers on the card don't fire
+        game.getState().processAction(game);
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/y/YarusRoarOfTheOldGods.java:113]: No needs here cause moveCards to battlefield already call applyeffect/processaction inside:

IMG_9623

Comment (DominionSpy) [reply Mage.Sets/src/mage/cards/y/YarusRoarOfTheOldGods.java:113]: If I leave out that line, bringing [[Ponyback Brigade]] to the battlefield (for example) causes its trigger to fire twice (enters the battlefield or is turned face up). Is that right?

Comment (JayDi85) [reply Mage.Sets/src/mage/cards/y/YarusRoarOfTheOldGods.java:113]: Yes, its two different actions and must raise two events (etb as face down + face up):

> return it to the battlefield face down under its owner’s control if it’s a permanent card, then turn it face up.

Comment (JayDi85) [reply Mage.Sets/src/mage/cards/y/YarusRoarOfTheOldGods.java]: Bug with Pony related to “etb with face down” problem from #11733

**Pull request merged**